fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/BaseDataSourceMetadataSpec.groovy

```diff
@@ -20,82 +20,128 @@ import io.druid.timeline.DataSegment
+ *     </li>
+ * </ul>
+ */
+abstract class BaseDataSourceMetadataSpec extends Specification {
     @Shared
-    String tableName = TestDruidTableName.ALL_PETS.asName()
+    DateTimeZone currentTZ
 
     @Shared
-    String intervalString1
-    @Shared
-    String intervalString2
+    String tableName
     @Shared
-    String intervalString12
-
+    Map<String, Interval> intervals
     @Shared
-    Interval interval1
+    RangeSet<DateTime> rangeSet
     @Shared
-    Interval interval2
-    @Shared
-    Interval interval12
-
+    List<TestApiDimensionName> dimensions
     @Shared
-    RangeSet<DateTime> rangeSet12
-
+    List<TestApiMetricName> metrics
     @Shared
-    String version1
+    Integer binversion1
     @Shared
-    String version2
-
+    long size1
     @Shared
-    DataSegment segment1
-
+    long size2
     @Shared
-    DataSegment segment2
-
+    List<String> versions
     @Shared
-    DataSegment segment3
+    List<DataSegment> segments
 
-    @Shared
-    DataSegment segment4
+    def setupSpec() {
+        currentTZ = DateTimeZone.getDefault()
+        DateTimeZone.setDefault(DateTimeZone.UTC)
+        binversion1 = 9
+        size1 = 1024
+        size2 = 512
 
-    @Shared
-    List<DataSegment> segments
+        childSetupSpec()
+    }
 
-    @Shared
-    DateTimeZone currentTZ
+    def childSetupSpec() {}
 
-    @Shared
-    List<String> dimensions123
+    def shutdownSpec() {
+        DateTimeZone.setDefault(currentTZ)
+    }
 
-    @Shared
-    List<String> metrics123
+    String generateTableName() {
+        TestDruidTableName.ALL_PETS.asName()
+    }
 
-    def setupSpec() {
-        currentTZ = DateTimeZone.getDefault()
-        DateTimeZone.setDefault(DateTimeZone.UTC)
+    Map<String, Interval> generateIntervals() {
+        [
+                "interval1": Interval.parse("2015-01-01T00:00:00.000Z/2015-01-02T00:00:00.000Z"),
+                "interval2": Interval.parse("2015-01-02T00:00:00.000Z/2015-01-03T00:00:00.000Z"),
+                "interval12": Interval.parse("2015-01-01T00:00:00.000Z/2015-01-03T00:00:00.000Z")
+        ]
+    }
 
-        intervalString1 = "2015-01-01T00:00:00.000Z/2015-01-02T00:00:00.000Z"
-        intervalString2 = "2015-01-02T00:00:00.000Z/2015-01-03T00:00:00.000Z"
-        intervalString12 = "2015-01-01T00:00:00.000Z/2015-01-03T00:00:00.000Z"
-        interval1 = Interval.parse(intervalString1)
-        interval2 = Interval.parse(intervalString2)
-        interval12 = Interval.parse(intervalString12)
-        rangeSet12 = TreeRangeSet.create()
-        rangeSet12.add(Range.closedOpen(interval12.getStart(), interval12.getEnd()))
-        version1 = DateTimeFormat.fullDateTime().print(DateTime.now().minusDays(1))
-        version2 = DateTimeFormat.fullDateTime().print(DateTime.now())
+    RangeSet<DateTime> generateRangeSet() {
+        rangeSet = TreeRangeSet.create()
+        rangeSet.add(Range.closedOpen(intervals["interval12"].getStart(), intervals["interval12"].getEnd()))
```

Comment:
Similarly, when accessing string keys in a map, you can just write `map.key`:

```groovy
        rangeSet.add(Range.closedOpen(intervals.interval12.getStart(), intervals.intervals12.getEnd
```
Though since you're pulling out interval12 twice, I'd probably pull it out into a variable:

```groovy
       Interval interval12 = intervals.interval12
```

Reply:
@archolewa Thank you very much for the tip!